pkg/network/tracer/testutil/testdns/test_dns_server.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package testdns
+
+import (
+	"net"
+	"os/exec"
+	"sync"
+	"testing"
+
+	"github.com/miekg/dns"
+	"github.com/stretchr/testify/require"
+)
+
+var globalServer *server
+var globalServerError error
+var serverOnce sync.Once
+
+// GetServerIP returns the IP address of the test DNS server. The test DNS server returns canned responses for several
+// known domains that are used in integration tests.
+//
+// see server#start to see which domains are handled.
+func GetServerIP(t *testing.T) net.IP {
+	serverOnce.Do(func() {
+		globalServer, globalServerError = NewServer()
+		globalServer.Start("tcp")
```

Comment:
should the local server only listen on the transport type that the test is validating? seems like it could be nice to fail if the test is trying to validate TCP DNS but is mistakenly using UDP

Reply:
This means every test needs to manage the lifecycle of the server instead of having a singleton.  Given 8.8.8.8 behaved similarly, and having a single server can speed up our (already very slow tests) I have a light preference towards keeping it this way.  Though that would be a pretty easy change.

Reply:
ah I forgot it was a singleton... yeah I think this way makes more sense then